### PR TITLE
Update getting starting page with a minimal example & link to flytesnacks

### PR DESCRIPTION
**Pull request by katrogan**
https://github.com/lyft/flyte/issues/677

### CODE REVIEW

File: rsts/user/getting_started/create_first.rst [hunk 1]
```diff
@@ -37,89 +34,61 @@ The most basic Flyte primitive is a "task". Flyte Tasks are units of work that c
...
-Replace the values above with your registry username, password, and registry endpoint.
-
-You may need to change the ``IMAGE_NAME`` in the Makefile to reflect your namespace in the docker registry. (ex ``{{my docker username}}/myflyteproject``)
-
-With the image built, we just need to register the tasks and workflows. The process is the same as what we had done previously. ::
-
-  docker run --network host -e FLYTE_PLATFORM_URL='127.0.0.1:30081' {{ your docker image }} pyflyte -p myflyteproject -d development -c sandbox.config register workflows
-
-After this, you should be able to visit the Flyte UI, and run the workflow as you did with ``flytesnacks`` previously.
+For detailed and interactive example workflow 'recipes' check out the `Flytesnacks Cookbook <https://flytecookbook.readthedocs.io/en/latest//>`_
```
Comment (wild-endeavor) [inline rsts/user/getting_started/create_first.rst:94]:
```suggestion
For detailed and interactive example workflow 'recipes' check out the `Flytesnacks Cookbook <https://flytecookbook.readthedocs.io/en/latest/>`__
```

File: rsts/user/getting_started/create_first.rst [hunk 2]
```diff
@@ -14,18 +14,15 @@ You can save some effort by cloning the ``flytesnacks`` repo, and re-initializin
...
-  rm single_step/edges.py
+Let's take a look at how easy it is to use Flyte by writing a task from scratch.
 
 Creating a Project
 ******************
 
 In Flyte, workflows are organized into namespaces called "Projects". When you register a workflow, it must be registered under a project.
 
-Lets create a new project called ``myflyteproject``. Use the project creation endpoint to create the new project ::
+For example, to create a new project called ``myflyteproject``: use the project creation endpoint to create the new project ::
```
Comment (kumare3) [inline rsts/user/getting_started/create_first.rst:25]: create a project should be flyte-cli, can we get rid of the curl command?

File: rsts/user/getting_started/create_first.rst [hunk 3]
```diff
@@ -37,89 +34,61 @@ The most basic Flyte primitive is a "task". Flyte Tasks are units of work that c
...
-  
-          img = cv2.imread(plane_fname, 0)
-          edges = cv2.Canny(img, 50, 200)  # hysteresis thresholds
-  
-          output_file = '{}/output.jpg'.format(tmpdir.name)
-          cv2.imwrite(output_file, edges)
-  
-          parsed_image.set(output_file)
+  @task
+  def edge_detection_canny(image_location:str) -> FlyteFile:
```
Comment (kumare3) [inline rsts/user/getting_started/create_first.rst:55]: I would like the image_location to be a Flytefile ?

Comment (kumare3) [reply rsts/user/getting_started/create_first.rst:55]: i guess we can do that as a follow up

File: rsts/user/getting_started/create_first.rst [hunk 4]
```diff
@@ -37,89 +34,61 @@ The most basic Flyte primitive is a "task". Flyte Tasks are units of work that c
...
+      return FlyteFile["jpg"](path=output_file)
 
 
 Some of the new concepts demonstrated here are:
 
-* ``wf_params``: The first argument to a python task is a Flyte SDK defined object that offers handlers like logging.
-* Inputs and outputs are first defined in the decorator, and then passed into the argument of the function.  Note that the names in the function signature need to match those in the decorator arguments.
-* A ``Blob`` is a Flyte Kit type that represents binary data.  It is used to offload data to a storage location like S3.  Here we use it to store an image.
+* Use the ``@task`` decorator to convert your typed python function to a Flyte task.
+* A ``FlyteFile`` is a Flyte Kit type that represents binary data.  It is used to offload data to a storage location like S3.  Here we use it to store an image.
```
Comment (kumare3) [inline rsts/user/getting_started/create_first.rst:74]: I guess we can use intersphinx here?

File: rsts/user/getting_started/create_first.rst [hunk 5]
```diff
@@ -37,89 +34,61 @@ The most basic Flyte primitive is a "task". Flyte Tasks are units of work that c
...
+
+      return FlyteFile["jpg"](path=output_file)
 
 
 Some of the new concepts demonstrated here are:
 
-* ``wf_params``: The first argument to a python task is a Flyte SDK defined object that offers handlers like logging.
-* Inputs and outputs are first defined in the decorator, and then passed into the argument of the function.  Note that the names in the function signature need to match those in the decorator arguments.
-* A ``Blob`` is a Flyte Kit type that represents binary data.  It is used to offload data to a storage location like S3.  Here we use it to store an image.
+* Use the ``@task`` decorator to convert your typed python function to a Flyte task.
```
Comment (kumare3) [inline rsts/user/getting_started/create_first.rst:73]: here too intersphinx

File: rsts/user/getting_started/create_first.rst [hunk 6]
```diff
@@ -37,89 +34,61 @@ The most basic Flyte primitive is a "task". Flyte Tasks are units of work that c
...
+          data = d.read()
+          opfile.write(data)
+
+      img = cv2.imread(plane_fname, 0)
+      edges = cv2.Canny(img, 50, 200)  # hysteresis thresholds
+
+      output_file = '{}/output.jpg'.format(working_dir.name)
+      cv2.imwrite(output_file, edges)
+
+      return FlyteFile["jpg"](path=output_file)
```
Comment (kumare3) [inline rsts/user/getting_started/create_first.rst:68]: can we say that you can iterate on this code by running the example

edge_detection_canny(image_location-"")..

File: rsts/user/getting_started/create_first.rst (hunk 2 above)
Comment (katrogan) [reply rsts/user/getting_started/create_first.rst:25]: sure, done @kumare3

File: rsts/user/getting_started/create_first.rst [hunk 7]
```diff
@@ -37,89 +34,61 @@ The most basic Flyte primitive is a "task". Flyte Tasks are units of work that c
...
-Replace the values above with your registry username, password, and registry endpoint.
-
-You may need to change the ``IMAGE_NAME`` in the Makefile to reflect your namespace in the docker registry. (ex ``{{my docker username}}/myflyteproject``)
-
-With the image built, we just need to register the tasks and workflows. The process is the same as what we had done previously. ::
-
-  docker run --network host -e FLYTE_PLATFORM_URL='127.0.0.1:30081' {{ your docker image }} pyflyte -p myflyteproject -d development -c sandbox.config register workflows
-
-After this, you should be able to visit the Flyte UI, and run the workflow as you did with ``flytesnacks`` previously.
+For detailed and interactive example workflow 'recipes' check out the `Flytesnacks Cookbook <https://flytecookbook.readthedocs.io/en/latest/>`__
```
Comment (kumare3) [inline rsts/user/getting_started/create_first.rst:96]: Dive deeper and try out more examples then look into. Also I think this should be the first `getting started` page, the one about installing flyte etc should come after this

Thus you might want to say, 
`Intrigued, want to learn more and try more complex examples, cookbook link`

File: rsts/user/getting_started/create_first.rst [hunk 8]
```diff
@@ -14,18 +14,15 @@ You can save some effort by cloning the ``flytesnacks`` repo, and re-initializin
   git init
   cd python
```
Comment (kumare3) [inline rsts/user/getting_started/create_first.rst:15]: do we need to clone? and this is wrong if so

what about 
Step 1: lets create a new folder
`` mkdir my_example``
2. Create this structure
    my_example/workflows
3.  pip install flytekit - `pip install flytekit==0.16.0b1`
4. now edit a file named
`my_example/workflows/x.py`

AS a follow up @wild-endeavor maybe we can have a dummy repo in flyteorg, flytekit starter template project?
then we can git-clone.

Then in the next section, `want to get this code running on a flyte deployment?`
1. Setup a sandbox deployment on your local machine
2. ,,,
a. Add a dockerfile
b. Add a config
c. run a commnad
d. run from ui

Comment (wild-endeavor) [reply rsts/user/getting_started/create_first.rst:15]: yeah, I like this structure.  We should eventually do a template of some kind yes.  I'll look into a couple options early next week and pick one.

File: rsts/user/getting_started/create_first.rst (hunk 5 above)
Comment (katrogan) [reply rsts/user/getting_started/create_first.rst:75]: done

File: rsts/user/getting_started/create_first.rst (hunk 7 above)
Comment (katrogan) [reply rsts/user/getting_started/create_first.rst]: done.

File: rsts/user/getting_started/create_first.rst (hunk 4 above)
Comment (katrogan) [reply rsts/user/getting_started/create_first.rst]: done

File: rsts/user/getting_started/create_first.rst [hunk 9]
```diff
@@ -37,89 +34,61 @@ The most basic Flyte primitive is a "task". Flyte Tasks are units of work that c
...
-      edges = Output(run_edge_detection.outputs.parsed_image, sdk_type=Types.Blob)
-  
+   @workflow
+   class EdgeDetectorWf(image_input: str) -> FlyteFile:
+       edges = edge_detection_canny(image_location=image_input)
+       return edges
+
 This code block creates a workflow, with one task. The workflow itself has an input (the link to an image) that gets passed into the task, and an output, which is the processed image.
 
+You can call this workflow ``EdgeDetectorWf(image_input=...)`` and iterate locally before moving on to register it with Flyte.
```
[the remark this answers is not in the document]
Comment (katrogan) [reply rsts/user/getting_started/create_first.rst:95]: done

File: rsts/user/getting_started/create_first.rst (hunk 6 above)
Comment (katrogan) [reply rsts/user/getting_started/create_first.rst:70]: done.

File: rsts/user/getting_started/create_first.rst (hunk 8 above)
Comment (katrogan) [reply rsts/user/getting_started/create_first.rst:15]: yes i'd like to have a working/copy-pastable example for the local sandbox story

**Comment by katrogan** [comment]
PTAL @kumare3

File: rsts/user/getting_started/index.rst [hunk 10]
```diff
@@ -12,6 +12,6 @@ Getting Started
    :maxdepth: 1
    :caption: Getting Started
 
-   examples
    create_first
+   examples
```
Comment (kumare3) [inline rsts/user/getting_started/index.rst:16]: ?

Comment (katrogan) [reply rsts/user/getting_started/index.rst:16]: you asked to change the order?

File: rsts/user/getting_started/create_first.rst [hunk 11]
```diff
@@ -4,122 +4,101 @@
 Writing Your First Workflow
 ########################################
 
+By the end of this getting started guide you'll be familiar with how easy it is to author a Flyte workflow, run and then deploy it locally.
```
Comment (kumare3) [inline rsts/user/getting_started/create_first.rst:7]:
```suggestion
By the end of this getting started guide you'll be familiar with how easy it is to author a Flyte workflow, run and then deploy it to a sandbox cluster (which can be run locally or on any kubernetes cluster).
```

File: rsts/user/getting_started/create_first.rst [hunk 12]
```diff
@@ -4,122 +4,101 @@
...
 
-You can save some effort by cloning the ``flytesnacks`` repo, and re-initializing it as a new git repository ::
+You can save some effort by cloning the ``flytekit-python-template`` repo, and re-initializing it as a new git repository ::
 
-  git clone git@github.com:lyft/flytesnacks.git myflyteproject
+  git clone git@github.com:lyft/flytekit-python-template.git myflyteproject
   cd myflyteproject
   rm -rf .git
   git init
   cd python
```
Comment (kumare3) [inline rsts/user/getting_started/create_first.rst:17]:
```suggestion
  cd myflyteproject
```

File: rsts/user/getting_started/create_first.rst (hunk 3 above)
Comment (kumare3) [reply rsts/user/getting_started/create_first.rst:46]: are we keeping this opencv example?

File: rsts/user/getting_started/create_first.rst [hunk 13]
```diff
@@ -4,122 +4,101 @@
...
 Some of the new concepts demonstrated here are:
 
-* ``wf_params``: The first argument to a python task is a Flyte SDK defined object that offers handlers like logging.
-* Inputs and outputs are first defined in the decorator, and then passed into the argument of the function.  Note that the names in the function signature need to match those in the decorator arguments.
-* A ``Blob`` is a Flyte Kit type that represents binary data.  It is used to offload data to a storage location like S3.  Here we use it to store an image.
+* Use the :py:func:`flytekit.task` decorator to convert your typed python function to a Flyte task.
+* A :py:class:`flytekit.types.file.FlyteFile` is a Flytekit type that represents binary data.  It is used to offload data to a storage location like S3.  Here we use it to store an image.
+
+
+You can call this task ``edge_detection_canny(image_location="https://images.unsplash.com/photo-1512289984044-071903207f5e?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=2250&q=80")`` and iterate locally before adding it to part of a larger overall workflow.
```
Comment (kumare3) [inline rsts/user/getting_started/create_first.rst:68]: better to make this a 
```rst
.. code-block::
```

File: rsts/user/getting_started/create_first.rst [hunk 14]
```diff
@@ -4,122 +4,101 @@
...
-************************
-
-Flyte fulfills tasks using docker images. You'll need to build a docker image from this code before it can run in Flyte. The repo has a make target to build the docker image for you ::
-
-  eval $(minikube -p minikube docker-env)  # If using the flyte sandbox via minikube
-  make docker_build
+This code block creates a workflow, with one task. The workflow itself has an input (the link to an image) that gets passed into the task, and an output, which is the processed image.
 
-If you have the flyte sandbox installed on your local machine, the image will be accessible to your Flyte system. If you're running a remote Flyte instance, you'll need to upload this image to a remote registry such as Dockerhub, Amazon ECR, or Google Container Registry, so that it can be used by the Flyte system.
+You can call this workflow ``EdgeDetectorWf(image_input=...)`` and iterate locally before moving on to register it with Flyte.
```
Comment (kumare3) [inline rsts/user/getting_started/create_first.rst:85]: same here, reason this is easy copy-pastable

File: rsts/user/getting_started/create_first.rst (hunk 12 above)
Comment (katrogan) [reply rsts/user/getting_started/create_first.rst:17]: i'll just remove. we should already be in myflyteproject

**Pull request closed without merging**